### PR TITLE
openthread: samples: Remove old THCI file.

### PR DESCRIPTION
**Pull request by canisLupus1313**
In order to unify the interface of Thread test GRL this commit removes `nRF_Connect_SDK.py`
and points end user to start using the unified file form openthread sub
module.

Signed-off-by: Przemyslaw Bida <przemyslaw.bida@nordicsemi.no>

### CODE REVIEW

**Comment by edmont** [comment]
This will conflict with https://github.com/nrfconnect/sdk-nrf/pull/6519

I'd suggest to rebase once that one is merged.

File: doc/nrf/ug_thread_certification.rst [hunk 1]
```diff
@@ -77,11 +77,37 @@ Complete the following steps to prepare for the certification tests:
...
+
+      to
+
+      .. code-block::
+         """
+            >> Thread Host Controller Interface
+            >> Device : nRF_Connect_SDK THCI
+            >> Class : nRF_Connect_SDK
+         """
+
```
Comment (wiba-nordic) [inline doc/nrf/ug_thread_certification.rst:101]:
```suggestion

      and

```
I think that'll make it a little clearer, but will need to see the built documentation to make sure if this needs further clarification.

File: doc/nrf/ug_thread_certification.rst [hunk 2]
```diff
@@ -77,11 +77,37 @@ Complete the following steps to prepare for the certification tests:
...
 
-   a. Copy the provided :file:`ncs/nrf/samples/openthread/cli/harness/nRF_Connect_SDK.py` file into :file:`C:\\GRL\\Thread1.1\\Thread_Harness\\THCI\\`.
+   a. Copy the provided :file:`ncs/modules/lib/openthread/tools/harness-thci/OpenThread.py` file into :file:`C:\\GRL\\Thread1.1\\Thread_Harness\\THCI\\nRF_Connect_SDK.py`.
 
    b. Copy the provided :file:`ncs/nrf/samples/openthread/cli/harness/nRF_Connect_SDK.jpg` file into :file:`C:\\GRL\\Thread1.1\\Web\\images\\`.
 
-   c. Edit :file:`C:\\GRL\\Thread1.1\\Web\\data\\deviceInputFields.xml` and prepend the following code:
+   c. Edit :file:`C:\\GRL\\Thread1.1\\Thread_Harness\\THCI\\nRF_Connect_SDK.py` as follows:
+
+      .. code-block::
```
Comment (wiba-nordic) [inline doc/nrf/ug_thread_certification.rst:86]: Try using this:
```suggestion
      .. code-block:: python
```
to get rid of the build error, though not 100% sure on that. Also applied to line 95 if it works.

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/ug_thread_certification.rst (hunk 1 above)
Comment (canisLupus1313) [reply doc/nrf/ug_thread_certification.rst:101]: fixed

File: doc/nrf/ug_thread_certification.rst (hunk 2 above)
Comment (canisLupus1313) [reply doc/nrf/ug_thread_certification.rst]: I commited Your suggestion thanks.

**Review by edmont — APPROVED** [review]

File: doc/nrf/ug_thread_certification.rst [hunk 3]
```diff
@@ -77,11 +77,35 @@ Complete the following steps to prepare for the certification tests:
...
+
+      and
+
+      .. code-block:: python
+         class OpenThread(OpenThreadTHCI, IThci):
+
+      to
+
+      .. code-block:: python
+         class nRF_Connect_SDK(OpenThreadTHCI, IThci):
```
Comment (wiba-nordic) [inline doc/nrf/ug_thread_certification.rst:106]:
```suggestion
      .. code-block:: python
      
            >> Thread Host Controller Interface
            >> Device : OpenThread THCI
            >> Class : OpenThread

      to

      .. code-block:: python
      
            >> Thread Host Controller Interface
            >> Device : nRF_Connect_SDK THCI
            >> Class : nRF_Connect_SDK

      and

      .. code-block:: python
      
         class OpenThread(OpenThreadTHCI, IThci):

      to

      .. code-block:: python
      
         class nRF_Connect_SDK(OpenThreadTHCI, IThci):
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]
Forgot the need for empty lines.

File: doc/nrf/ug_thread_certification.rst (hunk 3 above)
Comment (canisLupus1313) [reply doc/nrf/ug_thread_certification.rst]: works now

**Review by wiba-nordic — APPROVED** [review]

**Review by lmaciejonczyk — APPROVED** [review]

**Pull request merged**